packages/ckeditor5-find-and-replace/src/findandreplaceediting.ts

```diff
@@ -80,7 +80,11 @@ function onDocumentChange(
 				results.remove( markerName );
 			}
 
-			writer.removeMarker( markerName );
+			// This condition is checked in several tests anyway and seems to be ignored by instabul.
```

Comment:
It's not falsy in this scenario. It crashes when `results` object contain marker that no longer exists in document (because it was removed by the user in editor).

Reply:
I agree with @Dumluregn, we should remove `istanbul ignore`.

Currently, the tests do not fall within this `if` statement, and adding the `istanbul ignore` comment should not be the solution. Perhaps you will need to prepare mocks or additional API calls to simulate this scenario in unit tests. If you can replicate this scenario in manual tests, it should also be feasible in unit tests.
Please take some time to determine how you can prepare unit tests to cover this case.

Reply:
@DawidKossowski The problem is that tests[1] actually fall into this condition and if we remove it then they will simply fail. I added to the `Writer` class specified method[2] that actually does the same as current line and coverage is back equal 100% in this file. I suspect that instabul incorrectly handles the exception raised by `writer.removeMarker` and does not include this line into coverage at all. 

[1] `hitting "Find" with no result should watch document modifications and update highlighted item if not present`

[2] 
```ts 
public removeMarkerIfPresent( markerOrName: string | Marker ): void {
		this._assertWriterUsedCorrectly();

		const name = typeof markerOrName == 'string' ? markerOrName : markerOrName.name;

		if ( !this.model.markers.has( name ) ) {
			return;
		}

		const marker = this.model.markers.get( name )!;

		if ( !marker.managedUsingOperations ) {
			this.model.markers._remove( name );

			return;
		}

		const oldRange = marker.getRange();

		applyMarkerOperation( this, name, oldRange, null, marker.affectsData );
	}
```